src/markdown-pages/collect-data/opentelemetry/instrument.mdx

```diff
@@ -0,0 +1,404 @@
+
+```jsx fileName=app/Program.cs
+var builder = WebApplication.CreateBuilder(args);
+
+builder.Services.AddControllers();
+builder.Services.AddEndpointsApiExplorer();
+builder.Services.AddSwaggerGen();
+
+var app = builder.Build();
+
+if (app.Environment.IsDevelopment())
+{
+    app.UseSwagger();
+    app.UseSwaggerUI();
+}
+
+app.UseHttpsRedirection();
+
+app.UseAuthorization();
+
+app.MapControllers();
+
+app.Run();
+```
+
+</Project>
+
+<Callout variant="course" title="lab">
+
+This procedure is part of a lab that teaches you how to instrument your application with OpenTelemetry.
+
+Each procedure in the lab builds upon the last, so make sure you've completed the last procedure, [_Set up your lab environment_](/collect-data/opentelemetry/set-up-env), before starting this one.
+
+</Callout>
+
+Your .NET application is running and is getting a lot of traffic. You want to instrument it with OpenTelemetry because it lets you avoid being locked into a single vendor, it's open source, and it's supported by many actors in the observability space.
+
+Here, you learn how easy it is to configure the [OpenTelemetry SDK](https://github.com/open-telemetry/opentelemetry-dotnet) to automatically collect metrics and traces from your application and send them to New Relic so you can analyze the results later.
+
+## Instrument your application
+
+<Steps>
+
+<Step>
+
+In the terminal window that's running your development server, press `<CTRL-C>`.
+
+You should see your server shut down. Now you can add some dependencies and update your app logic.
+
+</Step>
+
+<Step>
+
+Add the OpenTelemetry SDK and supporting packages:
+
+<>
+
+```bash
+dotnet add app package --prerelease OpenTelemetry
+dotnet add app package --prerelease OpenTelemetry.Instrumentation.AspNetCore
+dotnet add app package --prerelease OpenTelemetry.Exporter.OpenTelemetryProtocol
+dotnet add app package --prerelease OpenTelemetry.Extensions.Hosting
+```
+
+</>
+
+</Step>
+
+<Step>
+
+Create two environment variables that you use to configure your OpenTelemetry pipelines:
+
+<>
+
+```bash
+export OTEL_EXPORTER_OTLP_ENDPOINT=<REGION-BASED-OTLP-ENDPOINT>
+export NR_LICENSE_KEY=<YOUR-NEW-RELIC-LICENSE-KEY>
+```
+
+The [OpenTelemetry Protocol](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/protocol/otlp.md) (OTLP) endpoint is the url of our OpenTelemetry receiver. Your service sends data directly to New Relic through this endpoint. [We have a US endpoint and an EU endpoint](https://docs.newrelic.com/docs/more-integrations/open-source-telemetry-integrations/opentelemetry/opentelemetry-quick-start/#review-settings), so make sure you choose the appropriate one based on your region.
+
+There are several different types of API keys to choose from in New Relic that each serve a different purpose. To instrument your application with OpenTelemetry, you need a [license key](https://docs.newrelic.com/docs/apis/intro-apis/new-relic-api-keys/#ingest-license-key).
+
```

Comment:
I think we should add a callout here to remind the reader to replace the end point and licence key in the above to make sure they export the right variables. Reminder can be useful